packages/vscode/src/index.ts

```diff
@@ -70,29 +69,41 @@ export async function activate(ext: ExtensionContext) {
     }
     return
   }
-
   // now if the root is an array, then it is an empty array
-  const cwd = (root && !Array.isArray(root))
-    ? path.resolve(projectPath, root)
-    : projectPath
-
-  const contextLoader = await registerRoot(ext, status, cwd)
-  ext.subscriptions.push(
-    commands.registerCommand('unocss.reload', async () => {
-      log.appendLine('🔁 Reloading...')
-      if (contextLoader.contextsMap.get(cwd) === null) {
-        contextLoader.contextsMap.delete(cwd)
-        const hasConfig = await contextLoader.loadContextInDirectory(cwd)
-        if (hasConfig) {
-          registerAutoComplete(cwd, contextLoader, ext)
-          registerAnnotations(cwd, contextLoader, status, ext)
-          registerSelectionStyle(cwd, contextLoader)
-        }
-      }
-      contextLoader.reload()
-      log.appendLine('✅ Reloaded.')
-    }),
-  )
+
+  const cacheMap = new Set()
```

Comment:
We should also cache for file resolution, so we don't need to run `findUp` every time user navigate to the file.

Reply:
In addition, we could also listen to the fs on adding/removing the config file and purge the cache.